ADC_irq/Core/Src/main.c

```diff
@@ -97,34 +98,28 @@ int main(void)
 						//configuring the next channel to be read
 						ADC1->CHSELR = ADC_CHSELR_CHSEL18;
 						ADC1->CR 	|= ADC_CR_ADSTART;	//starts the ADC
+
+						//calculates v_ref - as in RM
+						aux = v_ref;
+						v_ref = (3 * (int32_t) (*VREFINT_CAL));
+						v_ref = (v_ref * 1000) / aux;	//*1000 so we have Vref in mV
+														//the calculated internal reference voltage is now stored in v_ref
 				break;
 				case ADC_CHSELR_CHSEL18:				//Internal temperature
 						temperature = ADC1->DR;			//we'll store the data read in temperature

```

Comment:
Start the conversion again here, so the program will keep doing the conversions in a loop

Reply:
Select the channel 17 and start the conversion**